src/rviz/default_plugin/marker_utils.cpp

```diff
@@ -76,4 +84,289 @@ MarkerBase* createMarker(int marker_type, MarkerDisplay* owner, DisplayContext*
   }
 }
 
+bool checkMarkerMsg(const visualization_msgs::Marker& marker, MarkerDisplay* owner)
+{
+  std::stringstream ss;
+
+  if(marker.action != visualization_msgs::Marker::ADD)
+    return true;
+
+  switch (marker.type) {
+  case visualization_msgs::Marker::ARROW:
+    ss << checkQuaternion(marker);
+    ss << checkScale(marker);
+    ss << checkColor(marker);
+    ss << checkPointsArrow(marker);
+    ss << checkColorsEmpty(marker);
+    ss << checkTextEmpty(marker);
+    ss << checkMeshEmpty(marker);
+    break;
+
+  case visualization_msgs::Marker::CUBE:
+  case visualization_msgs::Marker::CYLINDER:
+  case visualization_msgs::Marker::SPHERE:
+    ss << checkQuaternion(marker);
+    ss << checkScale(marker);
+    ss << checkColor(marker);
+    ss << checkPointsEmpty(marker);
+    ss << checkColorsEmpty(marker);
+    ss << checkTextEmpty(marker);
+    ss << checkMeshEmpty(marker);
+    break;
+
+
+  case visualization_msgs::Marker::LINE_STRIP:
+  case visualization_msgs::Marker::LINE_LIST:
+    ss << checkQuaternion(marker);
+    ss << checkScaleLineStripAndList(marker);
+    ss << checkPointsNotEmpty(marker);
+    ss << checkColors(marker);
+    ss << checkTextEmpty(marker);
+    ss << checkMeshEmpty(marker);
+    break;
+
+  case visualization_msgs::Marker::SPHERE_LIST:
+  case visualization_msgs::Marker::CUBE_LIST:
+  case visualization_msgs::Marker::TRIANGLE_LIST:
+    ss << checkQuaternion(marker);
+    ss << checkScale(marker);
+    ss << checkPointsNotEmpty(marker);
+    ss << checkColors(marker);
+    ss << checkTextEmpty(marker);
+    ss << checkMeshEmpty(marker);
+    break;
+
+  case visualization_msgs::Marker::POINTS:
+    ss << checkScalePoints(marker);
+    ss << checkPointsNotEmpty(marker);
+    ss << checkColors(marker);
+    ss << checkTextEmpty(marker);
+    ss << checkMeshEmpty(marker);
+    break;
+
+  case visualization_msgs::Marker::TEXT_VIEW_FACING:
+    ss << checkColor(marker);
+    ss << checkScaleText(marker);
+    ss << checkTextNotEmptyOrWhitespace(marker);
+    ss << checkPointsEmpty(marker);
+    ss << checkColorsEmpty(marker);
+    ss << checkMeshEmpty(marker);
+    break;
+
+  case visualization_msgs::Marker::MESH_RESOURCE:
+    ss << checkQuaternion(marker);
+    ss << checkColor(marker);
+    ss << checkScale(marker);
+    ss << checkPointsEmpty(marker);
+    ss << checkColorsEmpty(marker);
+    ss << checkTextEmpty(marker);
+
```

Comment:
Remove empty line
```suggestion
```

Reply:
Where possible, `std:string` was replaced with `char*` in d02db0b

Reply:
Formatting has been done in 63e1d6a

Reply:
I just talked to @v4hn again. I misunderstood your comment and will fix it by passing the `std::stringstream` to the checker functions.

Reply:
In 86c4ffa the return values of the checker functions are replaced by a reference to the `std::stringstream`. Furthermore, `tellp` is used to check if a comma has to be added before adding something to the error message and to check if there is an error at all. 

db705e2 just removes an unnecessary check for adding a comma.